File: Solutions/Medium/3719. Longest Balanced Subarray I/Solution.py
class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        result: int = 0
        n: int = len(nums)
        for i in range(0, n):
            seenDict: dict[int:int] = {}
            even: int = 0
            odd: int = 0
            for j in range(i, n):
                if(nums[j] in seenDict):
                    seenDict[nums[j]] += 1
                else:
                    if(nums[j] % 2 == 0):
                        even += 1
                    else:
                        odd += 1
                    seenDict[nums[j]] = 1
                if (odd == even):
                    result = max(result, j - i + 1)
        return result        

File: Solutions/Medium/3719. Longest Balanced Subarray I/Solution.cs
public class Solution {
    public int LongestBalanced(int[] nums) 
    {
        int n = nums.Length;
        int result = 0;
        for (int i = 0; i < n; i++){
            int even = 0;
            int odd = 0;
            Dictionary<int, int> dict = new Dictionary<int, int>();
            for (int j = i; j < n; j++)
            {
                if(dict.ContainsKey(nums[j])) {
                    dict[nums[j]]++;
                }
                else{
                    if (nums[j] % 2 == 0) even++;
                    else odd++;
                    dict[nums[j]] = 1;
                }
                if (odd == even){
                    result = Math.Max(result, j-i+1);
                }
            }
        }
        return result;
    }
}